### ISSUE TITLE
23.07.04(화) Issue

### ISSUE THREAD

**Issue by SaFD-00** [SaFD-00/2023_RCV_SUMMER_URP#2]
# <공부한 내용>
### 선형대수 관점
- 대부분 선형회귀는 방정식의 개수가 변수의 개수보다 많게 됨, 이는 열 벡터의 선형조합으로 표현됨
- 열벡터의 선형결합은 column space 를 만들어 낸다 선형회귀에 완전히 들어 맞는 solution이 없기 때문에 촤대한 비슷 한 벡터를 column space 내부에서 찾기 위해서 정사영을 이용한다. 이때 실제 해벡터와 정사영 벡터의 차를 error 로 생각할 수 있다.
### 2.확률 통계관점
- 모집단의 분포가 정규분포라고 가정하고 데이터를 가장 잘 설명할 수 있는 정규분포를 찾아야함 -> 평균과 표준편차를 변화 시키면서 찾음
- likelihood function 이 최대가 되도록 하는 최적값 찾기
- 이때 선형회귀에 떨어져있는 데이터들 즉 노이즈들이 정규 분포라고 가정하고 이런 노이즈들을 가장 잘 설명할 수 있도록 해야한다.
### 3.미적분,최적화 관점
- 제약조건이 없을경우 convex 함수는 미분값이 0 인지점이 최소값
- mse 로 loss함수로 놓고 이를 벡터로 표현하게 되면 quadratic programming 을 통해 손실함수가 최소가 되도록 하는 파라미터를 찾을수 있음

## Numpy 로 back propagation 구현
- back propagation 을 구현 하기 위한 편미분,연쇄법칙,벡터미분,행렬미분 등 기초 개념 학습
- 덧셈,곱셈,ReLU,Sigmoid,Affine,SoftMax-CrossEntropy 연산에 대한 back propagation 연산구현
- 직접 구현한 back propagation 통해 실제 선형회귀 실습

# <공부할 부분>
- Customdata loader 익숙해지기
- Custom data loader 관련 document
(https://tutorials.pytorch.kr/beginner/basics/data_tutorial.html)
